src/components/PostDetails.tsx

```diff
@@ -1,106 +1,163 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Post } from '../types/Post';
+import { deleteComment, getComments, postComment } from '../services/post';
 import { Loader } from './Loader';
+import { Comment, CommentData } from '../types/Comment';
 import { NewCommentForm } from './NewCommentForm';
 
-export const PostDetails: React.FC = () => {
+interface Props {
+  selectedPost: Post;
+  isOpenedNewCommentForm: boolean;
+  onOpenNewCommentForm: (answer: boolean) => void;
+}
+
+export const PostDetails: React.FC<Props> = ({
+  selectedPost,
+  isOpenedNewCommentForm,
+  onOpenNewCommentForm,
+}) => {
+  const [comments, setComments] = useState<Comment[] | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [error, setError] = useState('');
+  const hasNoComments =
+    !isLoading && error.length === 0 && comments && comments.length === 0;
+  const hasAlreadyComments =
+    !isLoading && error.length === 0 && comments && comments.length > 0;
+  const isShowNewCommentButton =
+    !isLoading && !isOpenedNewCommentForm && error.length === 0;
+
+  const addNewComment = ({ name, email, body }: CommentData) => {
+    setIsSubmitting(true);
+
+    const postId = selectedPost.id;
+
+    return postComment({ postId, name, email, body })
+      .then(newComment => {
+        setComments(currentComments => {
+          if (currentComments) {
+            return [...currentComments, newComment];
+          }
+
+          return [newComment];
+        });
+      })
+      .catch(loadingError => {
+        setError('commentAddError');
+
+        throw loadingError;
+      })
+      .finally(() => {
+        setIsSubmitting(false);
+      });
+  };
+
+  const onDeleteComment = (commentId: number) => {
+    setComments(currentComments =>
+      currentComments
+        ? currentComments.filter(comment => comment.id !== commentId)
+        : [],
+    );
+
+    deleteComment(commentId).catch(() => {
+      setError('commentDeletingError');
+    });
+  };
+
+  useEffect(() => {
+    setIsLoading(true);
+    onOpenNewCommentForm(false);
+    setError('');
+
+    getComments(selectedPost.id)
+      .then(setComments)
+      .catch(loadingError => {
+        setError('commentsLoadingError');
```

Comment:
The error message 'commentsLoadingError' is being set as a string. Consider using a more descriptive error message or handling the error object to provide more context to the user.